arrays/arrays.c

```diff
@@ -114,33 +137,57 @@ void arr_append(Array *arr, char *element) {
  *
  * Throw an error if the value is not found.
  *****/
-void arr_remove(Array *arr, char *element) {
-
-  // Search for the first occurence of the element and remove it.
-  // Don't forget to free its memory!
-
-  // Shift over every element after the removed element to the left one position
+void arr_remove(Array *arr, char *element)
+{
+  int elementfound = 0;
+  int i = 0;
+  while (i < arr->count)
+  {
+    // Search for the first occurence of the element and remove it.
+    // Don't forget to free its memory!
+    if (elementfound == 0 && arr->elements[i] == element)
+    {
+      arr->elements[i] = NULL;
+      free(arr->elements[i]);
+      elementfound = 1;
```

Comment:
Awesome, I like how you use this `elementfound` variable to trigger certain conditions. It brings to mind the concept of flags which you will see more of next week in computer architecture. Very nicely done. 💯